contracts/cw-proposal-single/src/contract.rs

```diff
@@ -587,6 +645,36 @@ pub fn execute_remove_vote_hook(
         .add_attribute("address", address))
 }
 
+pub fn execute_assign_executor(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    address: Option<String>,
+) -> Result<Response, ContractError> {
+    let mut config = CONFIG.load(deps.storage)?;
+
+    // Only the DAO via prop OR the executor directly may call this method.
+    if info.sender == config.dao
+        || (config.executor_addr.is_some() && config.executor_addr.unwrap() == info.sender)
+    {
+        // executor can be removed or reassigned to another address
+        let new_executor_addr: Option<Addr> = match address {
+            None => None,
+            Some(addr_str) => Some(deps.api.addr_validate(&addr_str).unwrap()),
+        };
```

Comment:
Suggest doing same `.map().transpose()` maneuver as suggested above.

Reason being: generally speaking the people writing the standard library know a lot more about how to write good, high performance Rust than we do. Where we can use their stuff, we should. :)

Reply:
> 

good to know, ill keep that in mind next time!